algorithms/tarjan_scc.hpp

```diff
@@ -52,6 +52,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <unordered_map>
 #include <unordered_set>
 #include <vector>
+#include <map>
```

Comment:
From what I can tell TarjanSCC needs the following includes:

- `<cstddef>` for `std::size_t`
- `<vector>` for `std::vector`
- `<memory>` for `std::shared_ptr`
- `<stack>` for `std::stack`
- `<algorithm>` for `std::min`, `std::count_if`

but non of those:

- `<unordered_map>`
- `<unordered_set>`
- `<map>`